cmd/minikube/cmd/start.go

```diff
@@ -143,6 +147,20 @@ func runStart(cmd *cobra.Command, args []string) {
 		os.Setenv(k, v)
 	}
 
+	registrator, err := registration.DetectRegistrator(host.Driver)
```

Comment:
if you pull the whole registration code into a dedicated method it becomes testable. This way you can write tests for the various cases of parameters where the credentials are or are not specified. See also _start_test.go_